ckanext/s3filestore/views/uploads.py

```diff
@@ -40,9 +47,53 @@ def uploaded_file_redirect(upload_to, filename):
 
     return redirect(url)
 
+def signv4_upload():
+    aws_secret = ckan_config.get("ckanext.s3filestore.aws_secret_access_key", None)
+    region = ckan_config.get("ckanext.s3filestore.region_name", None)
+
+    to_sign = str(request.args.get('to_sign')).encode('utf-8')
```

Comment:
Probably need some verification on this -- that we're signing a request that the user should have access to. 
What's the authentication on this endpoint?

(for that matter, what's the authentication on the uploaded_file_redirect above?)

Reply:
Added an auth check for `signv4_upload()`

`uploaded_file_redirect(upload_to, filename)` will need to be open as that's what creates the redirect to download the file

Reply:
Can we check resource read on `upload_to`?  What I'm thinking here is if there's a private resource, this would essentially publish it.

Reply:
Looks like I was wrong, `uploaded_file_redirect(upload_to, filename)` was the previous method used to get a signed URL to the key, but I don't see how it was used.

With the addition of the `signv4_upload` the function and URL rule can be removed.